src/Datasource/QueryInterface.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.1
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Datasource;
+
+interface QueryInterface
+{
+
+    /**
+     * Returns a key => value array representing a single aliased field
+     * that can be passed directly to the select() method.
+     * The key will contain the alias and the value the actual field name.
+     *
+     * If the field is already aliased, then it will not be changed.
+     * If no $alias is passed, the default table for this query will be used.
+     *
+     * @param string $field The field to alias
+     * @param string $alias the alias used to prefix the field
+     * @return array
+     */
+    public function aliasField($field, $alias = null);
+
+    /**
+     * Returns the default repository object that will be used by this query,
+     * that is, the repository that will appear in the from clause.
+     *
+     * @param \Cake\Datasource\RepositoryInterface|null $repository The default repository object to use
+     * @return \Cake\Datasource\RepositoryInterface|$this
+     */
+    public function repository(RepositoryInterface $repository = null);
```

Comment:
I like that. I am wondering about ```where``` and ```order``` as those names is very focused on SQL. In case of the webservice plugin I've for example used ```conditions``` as ```where``` equivalent. I do understand if you'd like to call it ```where``` anyway.